com.unity.multiplayer.mlapi/Runtime/Core/NetworkUpdateLoopSystem.cs

```diff
@@ -0,0 +1,129 @@
+            return InternalRegisterNetworkUpdateStage(stage);
+        }
+
+        protected void RegisterUpdateLoopSystem()
+        {
+            NetworkUpdateManager.NetworkLoopRegistration(this);
+        }
+
+        protected void OnNetworkLoopSystemRemove()
+        {
+            if(onNetworkLoopSystemDestroyed != null)
+            {
+                onNetworkLoopSystemDestroyed.Invoke(this);
+            }
+        }
+
+        Action<INetworkUpdateLoopSystem> onNetworkLoopSystemDestroyed;
+
+        public void RegisterUpdateLoopSystemDestroyCallback(Action<INetworkUpdateLoopSystem> networkLoopSystemDestroyedCallback)
+        {
+            onNetworkLoopSystemDestroyed = networkLoopSystemDestroyedCallback;
+        }
+    }
+
+    /// <summary>
+    ///  UpdateLoopBehaviour
+    ///  Derive from this class if you only require MonoBehaviour functionality
+    /// </summary>
+    public class UpdateLoopBehaviour:MonoBehaviour,INetworkUpdateLoopSystem
+    {
+        protected virtual Action InternalRegisterNetworkUpdateStage(NetworkUpdateManager.NetworkUpdateStages stage )
+        {
+            return null;
+        }
+
+        public Action RegisterUpdate(NetworkUpdateManager.NetworkUpdateStages stage )
+        {
+            return InternalRegisterNetworkUpdateStage(stage);
+        }
+
+        protected void RegisterUpdateLoopSystem()
+        {
+            NetworkUpdateManager.NetworkLoopRegistration(this);
+        }
+
+        protected void OnNetworkLoopSystemRemove()
+        {
+            if(onNetworkLoopSystemDestroyed != null)
+            {
+                onNetworkLoopSystemDestroyed.Invoke(this);
+            }
+        }
+
+        Action<INetworkUpdateLoopSystem> onNetworkLoopSystemDestroyed;
+
+        public void RegisterUpdateLoopSystemDestroyCallback(Action<INetworkUpdateLoopSystem> networkLoopSystemDestroyedCallback)
+        {
+            onNetworkLoopSystemDestroyed = networkLoopSystemDestroyedCallback;
+        }
+    }
+
+    /// <summary>
+    /// GenericUpdateLoopSystem
+    /// Derive from this class for generic (non-MonoBehaviour) classes
+    /// </summary>
+    public class GenericUpdateLoopSystem:INetworkUpdateLoopSystem
+    {
+        protected virtual Action InternalRegisterNetworkUpdateStage(NetworkUpdateManager.NetworkUpdateStages stage )
+        {
+            return null;
+        }
+
+        public Action RegisterUpdate(NetworkUpdateManager.NetworkUpdateStages stage )
+        {
+            return InternalRegisterNetworkUpdateStage(stage);
+        }
+
+        protected void RegisterUpdateLoopSystem()
+        {
+            NetworkUpdateManager.NetworkLoopRegistration(this);
+        }
+
+        protected void OnNetworkLoopSystemRemove()
+        {
+            if(onNetworkLoopSystemDestroyed != null)
+            {
+                onNetworkLoopSystemDestroyed.Invoke(this);
+            }
+        }
+
+        Action<INetworkUpdateLoopSystem> onNetworkLoopSystemDestroyed;
```

Comment:
`private` access modifier to be explicit here?

Reply:
The coding convention, I just read specifically says not to verbosely add "private" it's ok to have it implicit. I have no opinion :-). I just state what is in the coding convention.